### PR TITLE
Added missing PSX arch COP2 instructions

### PR DESCRIPTION
**Pull request by PeterLemon**
Hi, this is my go at supporting the missing COP2 functionality of the PSX, allowing GTE programming.
Please see my PSX/GTE ARMIPS source code here for examples:
https://github.com/PeterLemon/PSX/tree/master/CPUTest/GTE

Let me know if you want me to change/improve anything =D

### CODE REVIEW

**Comment by MeganGrass** [comment]
Excellent! Is there a Windows binary with COP2 functionality available?

**Comment by PeterLemon** [comment]
Hi MeganGrass,
Yep I made a windows binary with cop2 functionality, here is link to it: https://expirebox.com/download/a22ae9b7b870e3920776aba2678630d2.html

File: Archs/MIPS/MipsOpcodes.cpp [hunk 1]
```diff
@@ -104,6 +105,8 @@ const tMipsOpcode MipsOpcodes[] = {
 	{ "lwc1",	"T,(s)",			MIPS_OP(0x31),			MA_MIPS1,	0 },
 	{ "l.s",	"T,i16(s)",			MIPS_OP(0x31),			MA_MIPS1,	0 },
 	{ "l.s",	"T,(s)",			MIPS_OP(0x31),			MA_MIPS1,	0 },
+	{ "lwc2",	"t,i16(s)",			MIPS_OP(0x32),			MA_PSX,	0 },
+	{ "lwc2",	"t,(s)",			MIPS_OP(0x32),			MA_PSX,	0 },
```
Comment (Kingcom) [inline Archs/MIPS/MipsOpcodes.cpp]: These are standard MIPS 1 opcodes, so they should be made available on all architectures that don't replace it with something else (i.e. MIPS1 except PSP).

File: Archs/MIPS/MipsOpcodes.cpp [hunk 2]
```diff
@@ -129,6 +132,8 @@ const tMipsOpcode MipsOpcodes[] = {
 	{ "swc1",	"T,(s)",			MIPS_OP(0x39),			MA_MIPS1,	0 },
 	{ "s.s",	"T,i16(s)",			MIPS_OP(0x39),			MA_MIPS1,	0 },
 	{ "s.s",	"T,(s)",			MIPS_OP(0x39),			MA_MIPS1,	0 },
+	{ "swc2",	"t,i16(s)",			MIPS_OP(0x3A),			MA_PSX,	0 },
+	{ "swc2",	"t,(s)",			MIPS_OP(0x3A),			MA_PSX,	0 },
```
Comment (Kingcom) [inline Archs/MIPS/MipsOpcodes.cpp]: Same here.

**Comment by PeterLemon** [comment]
Hi Kingcom,

I have added the PSX specific COP2 register names to my pull request.
I have also updated my github GTE src to use it (removed the register names from my GTE include file): https://github.com/PeterLemon/PSX/tree/master/CPUTest/GTE

Here is a build of the latest version for anyone that wants it:
https://expirebox.com/download/4e3ab4a5bfd0aff092b4d983b6e70c16.html

I read up on PS2 COP2, the "VPU0" is totally different to the PSX GTE, so I have chosen to not work on that as I have no experience with PS2 yet.

Let me know what you think =D

**Comment by Kingcom** [comment]
Looks good, thanks!

**Comment by PeterLemon** [comment]
Nice one, thanks Kingcom =D

**Pull request merged**